src/transformStateWithClones.js

```diff
@@ -8,6 +8,44 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(newState);
+        break;
+    }
```

Comment:
[CODE STYLE]: Use a `default` case in your `switch` statement for error handling. This will help catch any actions that do not match the expected types.